### DISCUSSION TITLE
zigbee2MQTT with sonoff -e USB

### DISCUSSION THREAD

**Discussion by Jhopkins1516** [Koenkk/zigbee2mqtt#16563] [Devices]
this is my first time setup on home assistant and I really want to make this work.
I am having trouble getting my zigbee2mqtt going.  this is running on an dell Wyse 5060 thin client. From what I see USB does show up. ZHA recognized it after pugging in but from my understanding z2m is the better setup. for the most part I cant even get past "502 bad gateway" 
I have made a mqtt-user and set up a password. 
I have configured core-mosquitto
I have checked to ensure passwords for user and in file are the same
my yaml is

data_path: /config/zigbee2mqtt
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: "-d -d"
  log: false
mqtt:
  server: mqtt://core-mosquitto:1883
  user: mqtt-user
  password: *********
serial:
  port: >-
    /dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20220817212213-if00
  adapter: ezsp

After all this my log shows the following has my first error.
MQTT error: Connection refused: Not authorized

Coordinator firmware version: '{"meta":{"maintrel":"3 ","majorrel":"6","minorrel":"10","product":8,"revision":"6.10.3.0 build 297"},"type":"EZSP v8"}'

**Comment by PortalRas** [comment]
I have the same problem.
I saw in a video that MQTT doesn't need to configure user and password because it uses the same as the home assistant automatically. But I'm in the same problem.

**Comment by PortalRas** [comment]
did you manage to resolve this issue?

**Comment by ddppddpp** [comment]
try troubleshooting talking to mosquitto as explained [here](https://www.home-assistant.io/integrations/mqtt/#testing-your-setup) or better yet start with the mosquitto log.

**Reply by PortalRas** [reply]
Hello, I tested my MQTT and everything is correct..

**Comment by Bahodirbm** [comment]
Hello, I have been through exact issues for the last 3 weeks and almost decided to give up on home assistant.  I can confirm that the issue on dongle e firmware. I have started flashing it with v6.7 (do not remember exact nr.) and i could get z2mqtt work. But my environment was unstable and my Ikea buttons ate batteries overnight. and then I flashed  v6.10 and all nightmares started again. Finally flashing dongle to v7.1 (not v7.2) I can confirt everything works well and stable. I still have issue with battery but it could be my battery quality as I used what I have been testing (not new from package). Now testing the whole system for 3rd day and so far all good!

**Reply by Jhopkins1516** [reply]
This could be a huge help I literally just flashed my stick to 6.10 last night and did not even start to configure mqtt. I just made sure everything reconnected to ZHA. I will look into 7.1 tonight. Thanks

**Reply by Bahodirbm** [reply]
I think OEM flashed 6.10 is not the same firmware v6.10 that you can download from Github. For me ZHA did not work well with IKEA controllers and moved to MQTT. After seeing what I can do in mqtt I am moving back to ZHA (not yet at least) :)

**Reply by nxgraphics** [reply]
where did you download the firmware for ZBDongle-E ? and where is the versioning:
https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP
versioning ?

**Reply by Bahodirbm** [reply]
I used different link, but I don't know if it's the same firmware.
https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle-NoSigned/EZSP

**Reply by nxgraphics** [reply]
your right, i also tried everything, all firmware updates, etc impossible to make work the dongle E with zigbee2mqtt...
from this page: https://www.zigbee2mqtt.io/guide/adapters/#not-recommended it is listed as Experimental...
i switched to **Zigbee Home Automation ( ZHA )** instead of zigbee2mqtt and it worked immediately..

**Comment by Bahodirbm** [comment]
Anyway, I gave up mqtt and dongle-e friendship. My Philips Hue dimmers failed several times and I moved back to ZHA now. Let's hope this Sonoff dongle -E works better with ZHA.
